forem/forem
# Add application deployment information to the admin page

## What type of PR is this? (check all applicable)


## Description
Forem administrators are now able to to determine from the admin page (http://localhost:3000/admin) when they last deployed and what was the last commit ID that was deployed.

I used the following to determine how to set the deploy and commit variables based on environment:
|                  | Forem Cloud Variables | Heroku Variables          |
|------------------|-----------------------|---------------------------|
| Last deployed    | FOREM_BUILD_DATE      | HEROKU_RELEASE_CREATED_AT |
| Latest Commit ID | FOREM_BUILD_SHA       | HEROKU_SLUG_COMMIT        |

I want to note that I did not use the RELEASE_FOOTPRINT which differs from what the `data-release-footprint` key in the `<body>` tag uses. Instead I used the FOREM_BUILD_DATE and FOREM_BUILD_SHA which seemed more apt. I could use some validation on this, and perhaps we could change the data keys in the body to reflect the same info as the admin page.

## Related Tickets & Documents
https://github.com/forem/InternalProjectPlanning/issues/230
Related slack thread: https://forem-team.slack.com/archives/C0134R988P9/p1607586614240100

## QA Instructions, Screenshots, Recordings
Add export HEROKU_RELEASE_CREATED_AT="Optional" to the env file to set the HEROKU_RELEASE_CREATED_AT variable. 

The reason that it says `Optional` under the values is because as per above comment I've set those values in the env file with the string Optional.
<img width="1680" alt="Screenshot 2021-01-06 at 12 35 14" src="https://user-images.githubusercontent.com/2786819/103758947-a22b0700-501b-11eb-8410-84d65f2509f6.png">

Besides changing the values for HEROKU_SLUG_COMMIT and HEROKU_RELEASE_CREATED_AT in the env file, I don't think there's a ton I can do to test this in development. Hence, I'd like to get the opinion of someone who is more knowledgeable than me on #systems and how these export variables are integrated into the app to ensure that this looks like it would work in prod. Based on this discussion I could put it behind a feature flag to test. 

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://docs.forem.com/frontend/accessibility)._

## Added tests?

      have not been included_

## Added to documentation?

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](gif_link)
